db/db_test_util.cc

```diff
@@ -428,6 +428,15 @@ Options DBTestBase::GetOptions(
       options.prefix_extractor.reset(NewNoopTransform());
       break;
     }
+    case kBlockBasedTableWithPartitionedIndexFormat3: {
+      table_options.index_type = BlockBasedTableOptions::kTwoLevelIndexSearch;
+      options.prefix_extractor.reset(NewNoopTransform());
+      // Format 3 changes the binary index format. Since partitioned index is a
+      // super-set of simple indexes, we are also using kTwoLevelIndexSearch to
+      // test this format.
+      table_options.format_version = 3;
```

Comment:
We need to be restraint about total test run time. Platforms like Travis has very tight duration limit. Test scenario added here will run a lot of tests with this new setting. I'm not against adding it but just be sure that you've thought of the question that whether we really want to test all of them with this new setting.

Reply:
Lets wait for travis run and verify the runtime.

Reply:
This is the comparison of runtime. The difference is not much and could actually be mostly noice:
Before:
```
 TEST_GROUP=platform_dependent
 19 min 46 sec
 TEST_GROUP=1
 30 min 31 sec
 TEST_GROUP=2
 22 min 57 sec
 TEST_GROUP=3
 18 min 9 sec
 TEST_GROUP=4
 14 min 38 sec
```
After:
```
 TEST_GROUP=platform_dependent
 16 min 21 sec
 TEST_GROUP=1
 33 min 34 sec
 TEST_GROUP=2
 19 min 54 sec
 TEST_GROUP=3
 23 min 18 sec
 TEST_GROUP=4
 15 min 33 sec
```